crates/ra_ide/src/diagnostics.rs

```diff
@@ -362,7 +371,12 @@ mod tests {
     }
 
     fn check_apply_diagnostic_fix(before: &str, after: &str) {
-        let (analysis, file_id) = single_file(before);
+        let (analysis, file_id) = if before.contains("//-") {
+            let (analysis, pos) = analysis_and_position(before);
+            (analysis, pos.file_id)
+        } else {
+            single_file(before)
+        };
         let diagnostic = analysis.diagnostics(file_id).unwrap().pop().unwrap();
```

Comment:
the test fails because there are no diagnostics, so I guess something is broken with the fixture handling.